src/librustc_mir/hair/pattern/_match.rs

```diff
@@ -931,12 +931,32 @@ impl<'tcx> IntRange<'tcx> {
     }
 }
 
-// Return a set of constructors equivalent to `all_ctors \ used_ctors`.
+// Used by `compute_missing_ctors`.
+#[derive(PartialEq)]
+enum Cost {
```

Comment:
I feel naming the variants of the function in terms of cost, rather than behaviour, is unhelpful — you can't guess what difference in behaviour might result from providing `Cheap` rather than `Expensive` (even if that is the motivation for the change). I'd rather stick with the comment about performance above `compute_missing_ctors` and give the variants more descriptive names.